### DISCUSSION TITLE
Data request failed with error: 'undefined' (25))

### DISCUSSION THREAD

**Discussion by MrNickert** [Koenkk/zigbee2mqtt#15336] [Devices]
Since yesterday all my bulbs are not responding, sometimes direct after a restart I can change a single bulb. All of them are hue. In the log files reading back the Data request failed with error: 'undefined' (25)).

See log below. How to resolve? 

`Zigbee2MQTT:error 2022-12-03 15:28:16: Publish 'set' 'state' to 'Achter tv links' failed: 'Error: Command 0x00178801033106e8/11 genOnOff.on({}, {"sendWhen":"immediate","timeout":10000,"disableResponse":false,"disableRecovery":false,"disableDefaultResponse":false,"direction":0,"srcEndpoint":null,"reservedBits":0,"manufacturerCode":null,"transactionSequenceNumber":null,"writeUndiv":false}) failed (Data request failed with error: 'undefined' (25))'
Zigbee2MQTT:info  2022-12-03 15:28:16: MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"message":"Publish 'set' 'state' to 'Achter tv links' failed: 'Error: Command 0x00178801033106e8/11 genOnOff.on({}, {\"sendWhen\":\"immediate\",\"timeout\":10000,\"disableResponse\":false,\"disableRecovery\":false,\"disableDefaultResponse\":false,\"direction\":0,\"srcEndpoint\":null,\"reservedBits\":0,\"manufacturerCode\":null,\"transactionSequenceNumber\":null,\"writeUndiv\":false}) failed (Data request failed with error: 'undefined' (25))'","meta":{"friendly_name":"Achter tv links"},"type":"zigbee_publish_error"}'`

**Comment by dafi87** [comment]
I am seeing this a lot in my network too now. Updating z2m from 1.28.2 to 1.28.4 did not solve it. It started happening after adding a Hue dimmer switch to the network.

**Comment by dafi87** [comment]
2022-12-07 12:56:26
Publish 'set' 'state' to 'Stehlampe oben' failed: 'Error: Command 0x001788070/11 genOnOff.off({}, {"sendWhen":"immediate","timeout":10000,"disableResponse":false,"disableRecovery":false,"disableDefaultResponse":false,"direction":0,"srcEndpoint":null,"reservedBits":0,"manufacturerCode":null,"transactionSequenceNumber":null,"writeUndiv":false}) failed (Data request failed with error: 'undefined' (25))'

2022-12-07 12:57:03
Failed to configure 'Schalter41', attempt 1 (Error: Read 0xcc86ef9a8a/1 genBasic(["manufacturerName","zclVersion","appVersion","modelId","powerSource",65534], {"sendWhen":"immediate","timeout":10000,"disableResponse":false,"disableRecovery":false,"disableDefaultResponse":true,"direction":0,"srcEndpoint":null,"reservedBits":0,"manufacturerCode":null,"transactionSequenceNumber":null,"writeUndiv":false}) failed (Data request failed with error: 'undefined' (25)) at ZStackAdapter.sendZclFrameToEndpointInternal (/opt/zigbee2mqtt/node_modules/zigbee-herdsman/src/adapter/z-stack/adapter/zStackAdapter.ts:415:23) at Queue.executeNext (/opt/zigbee2mqtt/node_modules/zigbee-herdsman/src/utils/queue.ts:32:32))

**Comment by jrhansen** [comment]
Yesterday, my mother in law's Zigbee network also broke down like that, leaving stuff like this in the logs:

```
error 2022-12-18 18:50:44: Publish 'set' 'brightness' to 'Køkken Lys, køkkenvask' failed: 'Error: Command 0x680ae2fffe947952/1 genLevelCtrl.moveToLevelWithOnOff({"level":128,"transtime":0}, {"sendWhen":"immediate","timeout":10000,"disableResponse":false,"disableRecovery":false,"disableDefaultResponse":false,"direction":0,"srcEndpoint":null,"reservedBits":0,"manufacturerCode":null,"transactionSequenceNumber":null,"writeUndiv":false}) failed (Data request failed with error: 'undefined' (25))'
info  2022-12-18 18:50:44: MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"message":"Publish 'set' 'brightness' to 'Køkken Lys, køkkenvask' failed: 'Error: Command 0x680ae2fffe947952/1 genLevelCtrl.moveToLevelWithOnOff({\"level\":128,\"transtime\":0}, {\"sendWhen\":\"immediate\",\"timeout\":10000,\"disableResponse\":false,\"disableRecovery\":false,\"disableDefaultResponse\":false,\"direction\":0,\"srcEndpoint\":null,\"reservedBits\":0,\"manufacturerCode\":null,\"transactionSequenceNumber\":null,\"writeUndiv\":false}) failed (Data request failed with error: 'undefined' (25))'","meta":{"friendly_name":"Køkken Lys, køkkenvask"},"type":"zigbee_publish_error"}'
error 2022-12-18 18:50:44: Publish 'set' 'brightness' to 'Køkken Lys, spisebord' failed: 'Error: Command 0x680ae2fffe4a0d48/1 genLevelCtrl.moveToLevelWithOnOff({"level":128,"transtime":0}, {"sendWhen":"immediate","timeout":10000,"disableResponse":false,"disableRecovery":false,"disableDefaultResponse":false,"direction":0,"srcEndpoint":null,"reservedBits":0,"manufacturerCode":null,"transactionSequenceNumber":null,"writeUndiv":false}) failed (Data request failed with error: 'undefined' (25))'
info  2022-12-18 18:50:44: MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"message":"Publish 'set' 'brightness' to 'Køkken Lys, spisebord' failed: 'Error: Command 0x680ae2fffe4a0d48/1 genLevelCtrl.moveToLevelWithOnOff({\"level\":128,\"transtime\":0}, {\"sendWhen\":\"immediate\",\"timeout\":10000,\"disableResponse\":false,\"disableRecovery\":false,\"disableDefaultResponse\":false,\"direction\":0,\"srcEndpoint\":null,\"reservedBits\":0,\"manufacturerCode\":null,\"transactionSequenceNumber\":null,\"writeUndiv\":false}) failed (Data request failed with error: 'undefined' (25))'","meta":{"friendly_name":"Køkken Lys, spisebord"},"type":"zigbee_publish_error"}'
error 2022-12-18 18:50:44: Publish 'set' 'color_temp' to 'Køkken Lys, køkkenvask' failed: 'Error: Command 0x680ae2fffe947952/1 lightingColorCtrl.moveToColorTemp({"colortemp":420,"transtime":0}, {"sendWhen":"immediate","timeout":10000,"disableResponse":false,"disableRecovery":false,"disableDefaultResponse":false,"direction":0,"srcEndpoint":null,"reservedBits":0,"manufacturerCode":null,"transactionSequenceNumber":null,"writeUndiv":false}) failed (Data request failed with error: 'undefined' (25))'
info  2022-12-18 18:50:44: MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"message":"Publish 'set' 'color_temp' to 'Køkken Lys, køkkenvask' failed: 'Error: Command 0x680ae2fffe947952/1 lightingColorCtrl.moveToColorTemp({\"colortemp\":420,\"transtime\":0}, {\"sendWhen\":\"immediate\",\"timeout\":10000,\"disableResponse\":false,\"disableRecovery\":false,\"disableDefaultResponse\":false,\"direction\":0,\"srcEndpoint\":null,\"reservedBits\":0,\"manufacturerCode\":null,\"transactionSequenceNumber\":null,\"writeUndiv\":false}) failed (Data request failed with error: 'undefined' (25))'","meta":{"friendly_name":"Køkken Lys, køkkenvask"},"type":"zigbee_publish_error"}'
error 2022-12-18 18:50:44: Publish 'set' 'color_temp' to 'Køkken Lys, spisebord' failed: 'Error: Command 0x680ae2fffe4a0d48/1 lightingColorCtrl.moveToColorTemp({"colortemp":422,"transtime":0}, {"sendWhen":"immediate","timeout":10000,"disableResponse":false,"disableRecovery":false,"disableDefaultResponse":false,"direction":0,"srcEndpoint":null,"reservedBits":0,"manufacturerCode":null,"transactionSequenceNumber":null,"writeUndiv":false}) failed (Data request failed with error: 'undefined' (25))'
info  2022-12-18 18:50:44: MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"message":"Publish 'set' 'color_temp' to 'Køkken Lys, spisebord' failed: 'Error: Command 0x680ae2fffe4a0d48/1 lightingColorCtrl.moveToColorTemp({\"colortemp\":422,\"transtime\":0}, {\"sendWhen\":\"immediate\",\"timeout\":10000,\"disableResponse\":false,\"disableRecovery\":false,\"disableDefaultResponse\":false,\"direction\":0,\"srcEndpoint\":null,\"reservedBits\":0,\"manufacturerCode\":null,\"transactionSequenceNumber\":null,\"writeUndiv\":false}) failed (Data request failed with error: 'undefined' (25))'","meta":{"friendly_name":"Køkken Lys, spisebord"},"type":"zigbee_publish_error"}'
error 2022-12-18 18:50:44: Publish 'set' 'brightness' to 'Forhave Lys, hoveddør' failed: 'Error: Command 0x5c0272fffe2a8b76/1 genLevelCtrl.moveToLevelWithOnOff({"level":254,"transtime":0}, {"sendWhen":"immediate","timeout":10000,"disableResponse":false,"disableRecovery":false,"disableDefaultResponse":false,"direction":0,"srcEndpoint":null,"reservedBits":0,"manufacturerCode":null,"transactionSequenceNumber":null,"writeUndiv":false}) failed (Data request failed with error: 'undefined' (25))'
info  2022-12-18 18:50:44: MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"message":"Publish 'set' 'brightness' to 'Forhave Lys, hoveddør' failed: 'Error: Command 0x5c0272fffe2a8b76/1 genLevelCtrl.moveToLevelWithOnOff({\"level\":254,\"transtime\":0}, {\"sendWhen\":\"immediate\",\"timeout\":10000,\"disableResponse\":false,\"disableRecovery\":false,\"disableDefaultResponse\":false,\"direction\":0,\"srcEndpoint\":null,\"reservedBits\":0,\"manufacturerCode\":null,\"transactionSequenceNumber\":null,\"writeUndiv\":false}) failed (Data request failed with error: 'undefined' (25))'","meta":{"friendly_name":"Forhave Lys, hoveddør"},"type":"zigbee_publish_error"}'
error 2022-12-18 18:50:44: Publish 'set' 'brightness' to 'Forhave Lys, skur' failed: 'Error: Command 0x00178801024968f9/11 genLevelCtrl.moveToLevelWithOnOff({"level":254,"transtime":0}, {"sendWhen":"immediate","timeout":10000,"disableResponse":false,"disableRecovery":false,"disableDefaultResponse":false,"direction":0,"srcEndpoint":null,"reservedBits":0,"manufacturerCode":null,"transactionSequenceNumber":null,"writeUndiv":false}) failed (Data request failed with error: 'undefined' (25))'
info  2022-12-18 18:50:44: MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"message":"Publish 'set' 'brightness' to 'Forhave Lys, skur' failed: 'Error: Command 0x00178801024968f9/11 genLevelCtrl.moveToLevelWithOnOff({\"level\":254,\"transtime\":0}, {\"sendWhen\":\"immediate\",\"timeout\":10000,\"disableResponse\":false,\"disableRecovery\":false,\"disableDefaultResponse\":false,\"direction\":0,\"srcEndpoint\":null,\"reservedBits\":0,\"manufacturerCode\":null,\"transactionSequenceNumber\":null,\"writeUndiv\":false}) failed (Data request failed with error: 'undefined' (25))'","meta":{"friendly_name":"Forhave Lys, skur"},"type":"zigbee_publish_error"}'
error 2022-12-18 18:50:57: Publish 'set' 'state' to 'Stue Lys, sofa, højre' failed: 'Error: Command 0x680ae2fffe36dd24/1 genOnOff.on({}, {"sendWhen":"immediate","timeout":10000,"disableResponse":false,"disableRecovery":false,"disableDefaultResponse":false,"direction":0,"srcEndpoint":null,"reservedBits":0,"manufacturerCode":null,"transactionSequenceNumber":null,"writeUndiv":false}) failed (Data request failed with error: 'undefined' (25))'
info  2022-12-18 18:50:57: MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"message":"Publish 'set' 'state' to 'Stue Lys, sofa, højre' failed: 'Error: Command 0x680ae2fffe36dd24/1 genOnOff.on({}, {\"sendWhen\":\"immediate\",\"timeout\":10000,\"disableResponse\":false,\"disableRecovery\":false,\"disableDefaultResponse\":false,\"direction\":0,\"srcEndpoint\":null,\"reservedBits\":0,\"manufacturerCode\":null,\"transactionSequenceNumber\":null,\"writeUndiv\":false}) failed (Data request failed with error: 'undefined' (25))'","meta":{"friendly_name":"Stue Lys, sofa, højre"},"type":"zigbee_publish_error"}'
```

We were still getting some data from sensors (temperature sensors etc.) and also some from the lights and switches, which we were no longer able to control.

I rebooted and power-cycled the RPi to no avail. When I checked the flash on the Sonoff Zigbee USB dongle, apparently there was a change in some of pages of the flash, so I reprogrammed it - again to no avail. I tried re-pairing one of the lights, but even that was not possible. As soon as the light left the network, Z2M could not recognize it anymore when trying to re-pair.

To me it is a complete mystery as to what happened. No one was even home, when it happened. Did anyone find a reason for this happening (or a fix)?

**Comment by jrhansen** [comment]
Also, this is the first error occurring in the logs. Don't know if it "means something" to someone smarter than me:

`error 2022-12-18 18:11:58: Error: CommandResponse 0xec1bbdfffec506cf/1 genOta.queryNextImageResponse({"status":152}, {"sendWhen":"immediate","timeout":10000,"disableResponse":false,"disableRecovery":false,"disableDefaultResponse":true,"direction":1,"srcEndpoint":null,"reservedBits":0,"manufacturerCode":null,"transactionSequenceNumber":null,"writeUndiv":false}) failed (Data request failed with error: 'undefined' (25))`

**Comment by Xythus** [comment]
Same issue here. My Zigbee network was solid for a few months prior to last night. Woke up to random devices not working and these kinds of messages in the logs:

`2023-01-25 08:50:49Publish 'set' 'state' to 'Living Room - Ceiling Light' failed: 'Error: Command 0x50325ffffe1a2b6b/1 genLevelCtrl.moveToLevelWithOnOff({"level":0,"transtime":10}, {"sendWhen":"immediate","timeout":10000,"disableResponse":false,"disableRecovery":false,"disableDefaultResponse":false,"direction":0,"srcEndpoint":null,"reservedBits":0,"manufacturerCode":null,"transactionSequenceNumber":null,"writeUndiv":false}) failed (Data request failed with error: 'undefined' (25))'`

**Comment by Koenkk** [comment]
What coordinator and firmware are you using?

**Comment by MrNickert** [comment]
Sonoff 3.0 stick. With latest firmware. Already re-flashed a couple of times.

**Reply by Koenkk** [reply]
Can you provide the date of that firmware? A new fw will be released soon: https://github.com/Koenkk/Z-Stack-firmware/tree/6.10.01.01/coordinator/Z-Stack_3.x.0/bin

**Comment by MrNickert** [comment]
CC1352P2_CC2652P_launchpad_router_20220125

**Reply by Koenkk** [reply]
Then please try `20221226`

**Reply by aik** [reply]
is there anything newer? 20221226 is no good :-/

**Comment by Xythus** [comment]
I'm using an Electrolama zzh running CC2652R_coordinator_20220219

**Comment by ssinghj** [comment]
Hi,
I was also having very solid zigbee network until i updated to V.1.30 since then I am also starting getting rendom error from multiple zigbee devices. Lights and motion sensor either  not working or working with significant delay.
I am using Z2M with deConz stick
````
Publish 'set' 'state' to 'sw_in_hall_way' failed: 'Error: Command 0x00124b00257b1915/1 genOnOff.off({}, {"sendWhen":"immediate","timeout":10000,"disableResponse":false,"disableRecovery":false,"disableDefaultResponse":false,"direction":0,"srcEndpoint":null,"reservedBits":0,"manufacturerCode":null,"transactionSequenceNumber":null,"writeUndiv":false}) failed (no response received)' 
````

**Comment by aik** [comment]
My network broke now just like that. I was using CC2531 which was working fine, a week ago upgraded to Sonoff-P on CC2652 and for a week it was patchy but sorta working and today it just broke completely.
I tried re-pairing 3 Ikea lights to CC2652 but pairing just failed, multiple times, and those lights worked just yesterday.

CC1352P2_CC2652P_launchpad_coordinator_20221226.zip runs on 2652.

```
Mar 25 12:25:56 mqtt npm[26868]: Zigbee2MQTT:error 2023-03-25 12:25:56: Publish 'set' 'state' to 'study_light' failed: 'Error: Command 0x847127fffed029cf/1 genOnOff.off({}, {"sendWhen":"immediate","timeout":10000,"disableResponse":false,"disableRecovery":false,"disableDefaultResponse":false,"direction":0,"srcEndpoint":null,"reservedBits":0,"manufacturerCode":null,"transactionSequenceNumber":null,"writeUndiv":false}) failed (Data request failed with error: 'undefined' (25))'
Mar 25 12:25:56 mqtt npm[26868]: Zigbee2MQTT:info  2023-03-25 12:25:56: MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"message":"Publish 'set' 'state' to 'study_light' failed: 'Error: Command 0x847127fffed029cf/1 genOnOff.off({}, {\"sendWhen\":\"immediate\",\"timeout\":10000,\"disableResponse\":false,\"disableRecovery\":false,\"disableDefaultResponse\":false,\"direction\":0,\"srcEndpoint\":null,\"reservedBits\":0,\"manufacturerCode\":null,\"transactionSequenceNumber\":null,\"writeUndiv\":false}) failed (Data request failed with error: 'undefined' (25))'","meta":{"friendly_name":"study_light"},"type":"zigbee_publish_error"}'
Mar 25 12:26:00 mqtt npm[26868]: Zigbee2MQTT:error 2023-03-25 12:26:00: Exception while calling fromZigbee converter: Command 0x804b50fffea4b2b3/1 manuSpecificTuya.mcuGatewayConnectionStatus({"payloadSize":1,"payload":1}, {"sendWhen":"immediate","timeout":10000,"disableResponse":false,"disableRecovery":false,"disableDefaultResponse":false,"direction":0,"srcEndpoint":null,"reservedBits":0,"manufacturerCode":null,"transactionSequenceNumber":null,"writeUndiv":false}) failed (Data request failed with error: 'undefined' (25))}

```
~                                                                                                                                              
  
Switching to the dev branch did not help.

For now I put cc2531 back and the things seems to be back to normal. I could even re-pair the 3 lights mentioned above. My cc2652 and cc2531 use the same mac so **no** re-pairing was done when I switched CC dongles (except those lights).

However even with cc2531 there are occasional 
`Mar 25 12:39:26 mqtt npm[27643]: Zigbee2MQTT:error 2023-03-25 12:39:26: Exception while calling fromZigbee converter: Command 0x804b50fffea4b2b3/1 manuSpecificTuya.mcuGatewayConnectionStatus({"payloadSize":1,"payload":1}, {"sendWhen":"immediate","timeout":10000,"disableResponse":false,"disableRecovery":false,"disableDefaultResponse":false,"direction":0,"srcEndpoint":null,"reservedBits":0,"manufacturerCode":null,"transactionSequenceNumber":null,"writeUndiv":false}) failed (Data request failed with error: 'undefined' (25))}`
which is from TS0601_clamp_meter, do not know if this is related - the sensor readings are mostly ok though.

Any chance there is a newer CC2652 fw?

**Comment by dafi87** [comment]
Is there any news on this subject? I'm still seeing requests fail on my network with the 'undefined' (25) error on a daily basis.
Does anybody know if there's already an open issue for this?

**Comment by MrNickert** [comment]
I moved away from zigbee2mqtt.

From: dafi87 ***@***.***>
Date: Friday, 5 May 2023 at 09:53
To: Koenkk/zigbee2mqtt ***@***.***>
Cc: Nick Meijer ***@***.***>, Author ***@***.***>
Subject: Re: [Koenkk/zigbee2mqtt] Data request failed with error: 'undefined' (25)) (Discussion #15336)

Is there any news on this subject? I'm still seeing requests fail on my network with the 'undefined' (25) error on a daily basis.
Does anybody know if there's already an open issue for this?
